### PR TITLE
fix(evpn-bridge): fix empty vni use cases for vrf and logical bridge

### PR DESCRIPTION
**Pull request by atulpatel261194**

### CODE REVIEW

File: pkg/infradb/infradb.go [hunk 1]
```diff
@@ -321,24 +321,25 @@ func UpdateLBStatus(name string, resourceVersion string, notificationID string,
...
 			if lb.Spec.Vni != nil {
-				delete(vpns, *lb.Spec.Vni)
-			}
-			err = infradb.client.Set("vpns", &vpns)
-			if err != nil {
-				log.Println(err)
-				return err
+				found, err = infradb.client.Get("vpns", &vpns)
+				if err != nil {
+					log.Println(err)
```
Comment (artek-koltun) [inline pkg/infradb/infradb.go:327]: is it a good strategy to log an error here and return it?
Apparently, a caller will likely also print the same error. Doing so, we are causing a lot of noise in the application logs for, in my opinion, little value.

File: pkg/infradb/infradb.go [hunk 2]
```diff
@@ -975,24 +976,25 @@ func UpdateVrfStatus(name string, resourceVersion string, notificationID string,
...
-			if err != nil {
-				log.Println(err)
-				return err
-			}
 
+				err = infradb.client.Set("vpns", &vpns)
+				if err != nil {
+					log.Println(err)
+					return err
+				}
```
Comment (artek-koltun) [inline pkg/infradb/infradb.go:996]: This chunk looks very similar to the one above...
Can we create a dedicated function to do that and eliminate the duplication? Maybe `removeVniFromVpns`?

File: pkg/infradb/infradb.go (hunk 1 above)
Comment (atulpatel261194) [reply pkg/infradb/infradb.go]: done

File: pkg/infradb/infradb.go (hunk 2 above)
Comment (atulpatel261194) [reply pkg/infradb/infradb.go:994]: Done

File: pkg/infradb/infradb.go [hunk 3]
```diff
@@ -257,6 +257,29 @@ func UpdateLB(lb *LogicalBridge) error {
...
+			log.Println("removeVniFromVpns(): No VPNs have been found")
+			log.Println(vni, ":")
+			return ErrKeyNotFound
+		}
+		delete(vpns, vni)
+
+		err = infradb.client.Set("vpns", &vpns)
+		if err != nil {
+			log.Println(err)
+			return err
```
Comment (artek-koltun) [inline pkg/infradb/infradb.go:278]: again, here we print an error msg and return that err.
On the level above, we print the same error. What value do we add by behaving that way? Probably we just flood the log with the identical error messages.
Each error should be handled once, and logging is a form of handling. You should either log it or return it, not both

**Review by artek-koltun — APPROVED** [review]

**Pull request merged**